fhir-persistence-schema/src/main/resources/postgres/delete_resource_parameters_sharded.sql

```diff
@@ -0,0 +1,65 @@
+-------------------------------------------------------------------------------
+-- (C) Copyright IBM Corp. 2021
+--
+-- SPDX-License-Identifier: Apache-2.0
+-------------------------------------------------------------------------------
+
+-- ----------------------------------------------------------------------------
+-- Procedure to delete all search parameters values for a given resource.
+-- This variant is for use with the distributed schema variant typically
+-- deployed to a distributed database service like Citus.
```

Comment:
how does it differ from delete_resource_parameters_distributed.sql ?

Reply:
The sharded variant needs to include the shard key as part of the delete:
	EXECUTE 'DELETE FROM {{SCHEMA_NAME}}.' || p_resource_type || '_str_values          WHERE shard_key = $1 AND logical_resource_id = $2'
	USING p_shard_key, p_logical_resource_id;

It is the _distributed variant we no longer need because it is identical to the plain variant.

Reply:
I've removed the citus directory of procedures entirely because it's no longer required. Currently the citus variant uses the same logic as PostgreSQL, but we may change this in the future when it comes time to performance tune ingestion - it's possible we'll need to distribute the functions too for better performance, but we'll cross that bridge when we come to it.